docs/writing-tests/unit-testing.md

```diff
@@ -0,0 +1,137 @@
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/component-test-with-testing-library.js.mdx',
+    'react/component-test-with-testing-library.ts.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+
+💡 You can use Testing Library out-of-the-box with [Storybook Interaction Testing](./interaction-testing.md).
+
+</div>
+
+Once the test runs, it loads the story and renders it. [Testing Library](https://testing-library.com/) then emulates the user's behavior and checks if the component state has been updated.
+
+## Configure
+
+By default, Storybook offers a zero-config setup for React and other frameworks via addons, allowing you to run your stories as tests with Testing Library. However, adding a configuration file can extend your test setup and include additional global [decorators](../writing-stories/decorators.md#global-decorators), [parameters](../writing-stories/parameters.md#global-parameters), and other features. To do so, create a `setup.js|ts` file as follows:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/storybook-testing-addon-optional-config.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+Update your test script to include the configuration file:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/storybook-testing-addon-optional-config-scripts.json.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+### Override story properties
+
+By default, the `setProjectAnnotations` function injects into your existing tests any global configuration you've defined in your Storybook instance (i.e., parameters, decorators in the `preview.js|ts` file). Nevertheless, this may cause unforeseen side effects for tests that are not intended to use these global configurations. To avoid this, you can override the global configurations by extending either the `composeStory` or `composeStories` functions to provide test-specific configurations. For example:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/override-compose-story-test.compose-story.js.mdx',
+    'react/override-compose-story-test.compose-story.ts.mdx',
+    'react/override-compose-story-test.compose-stories.js.mdx',
+    'react/override-compose-story-test.compose-stories.ts.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+## Run tests on a single story
+
+You can use the `composeStory` function from the appropriate framework or supported addon to allow your tests to run on a single story. However, if you're relying on this method, we recommend that you supply the story metadata (e.g., the [default export](../writing-stories/introduction.md#default-export)) to the `composeStory` function. This ensures that your tests can accurately determine the correct information about the story. For example:
```

Comment:
This is a little confusing. The Meta is required when using `composeStory(story, meta)`, so users have to always pass it